app/controllers/GalleryController.scala

```diff
@@ -0,0 +1,135 @@
+package controllers
+
+import javax.inject.Inject
+import java.sql.Timestamp
+import java.time.Instant
+
+import controllers.headers.ProvidesHeader
+import models.user._
+import models.label.LabelTable
+import models.label.LabelTable._
+import com.mohiva.play.silhouette.api.{Environment, Silhouette}
+import com.mohiva.play.silhouette.impl.authenticators.SessionAuthenticator
+import play.api.libs.json.{JsObject, Json, JsValue, JsArray}
+import play.api.Logger
+
+import scala.concurrent.Future
+import scala.io.Source
+
+
+class GalleryController @Inject() (implicit val env: Environment[User, SessionAuthenticator])
+  extends Silhouette[User, SessionAuthenticator] with ProvidesHeader {
+
+  // Set of valid labels
+  // TODO: Put somewhere more reasonable.
+  def validLabelIds: Set[Int] = Set(1, 2, 3, 4, 5, 6, 7)
+
+  /**
+   * Returns labels of a specified type.
+   *
+   * @param labelTypeId Label type specifying what type of labels to grab.
+   * @param n Number of labels to grab.
+   * @param loadedLabels String representing set of labelIds already grabbed as to not grab them again.
+   * @return
+   */
+  def getLabelsByType(labelTypeId: Int, n: Int, loadedLabels: String) = UserAwareAction.async { implicit request =>
+    request.identity match {
+      case Some(user) =>
+        val loadedLabelIds: Set[Int] = Json.parse(loadedLabels).as[JsArray].value.map(_.as[Int]).toSet
+        val labels: Seq[LabelValidationMetadata] = if (validLabelIds.contains(labelTypeId)) 
```

Comment:
I think that for Scala I'd prefer this indentation:
```
        val labels: Seq[LabelValidationMetadata] =
          if (validLabelIds.contains(labelTypeId)) LabelTable.retrieveLabelsByType(labelTypeId, n, loadedLabelIds) 
          else LabelTable.retrieveAssortedLabels(n, loadedLabelIds) 
```

Reply:
Sounds good

Reply:
There is one edge case that doesn't seem to fit the 120 character limit. Should I leave it like this?
```
val labels: Seq[LabelValidationMetadata] = 
          if (validLabelIds.contains(labelTypeId)) LabelTable.retrieveLabelsOfTypeBySeverityAndTags(labelTypeId, 
                                                                                                    n, 
                                                                                                    loadedLabelIds, 
                                                                                                    severitiesToSelect, 
                                                                                                    tagsToSelect)
          else LabelTable.retrieveAssortedLabels(n, loadedLabelIds, Some(severitiesToSelect))

Reply:
I'd be down with either
```
		val labels: Seq[LabelValidationMetadata] = 
          if (validLabelIds.contains(labelTypeId)) {
            LabelTable.retrieveLabelsOfTypeBySeverityAndTags(
              labelTypeId, n, loadedLabelIds, severitiesToSelect, tagsToSelect
            )
          }
          else {
            LabelTable.retrieveAssortedLabels(n, loadedLabelIds, Some(severitiesToSelect))
          }
```

or this
```
		val labels: Seq[LabelValidationMetadata] = 
          if (validLabelIds.contains(labelTypeId)) {
            LabelTable.retrieveLabelsOfTypeBySeverityAndTags(labelTypeId, n, loadedLabelIds, severitiesToSelect, tagsToSelect)
          }
          else {
            LabelTable.retrieveAssortedLabels(n, loadedLabelIds, Some(severitiesToSelect))
          }
```

or this
```
		val labels: Seq[LabelValidationMetadata] = 
          if (validLabelIds.contains(labelTypeId)) LabelTable.retrieveLabelsOfTypeBySeverityAndTags(labelTypeId, n, loadedLabelIds, severitiesToSelect, tagsToSelect)
          else LabelTable.retrieveAssortedLabels(n, loadedLabelIds, Some(severitiesToSelect))
```

I do like the style of having the parameters indented to the same place, but I just never really see that in Scala code (except for our case classes, but not sure we should be doing it there either).

Reply:
Wait the markdown is a little confusing. Should I be back indenting the lines that are a bit long? Or are those supposed to be indented one level in from the `val labels...` line?

Reply:
Okay I just fixed it. I pasted it into some other editor that I think turned some spaces into tabs, which messed up the look of the Markdown.

Reply:
Gotcha, I think the first one is the only one that meets the 120 line requirement so I'll do that.